backend/views/views.py

```diff
@@ -86,13 +86,39 @@ def user_modify(users_id):
         return render_template('user_modify.html', form=form, route_to=route_to)
 
 
-@app.route('/deleteuser/<int:users_id>')
+@app.route('/deleteuser/<int:users_id>', methods=['POST'])
+@admin_permissions
 def delete_user(users_id):
+    msg = "cannot delete user"
     current_moment = datetime.now()
     user = db.session.query(User).get(users_id)
-    user.delete_date = current_moment
+    if not is_last_admin(user):
+        user.delete_date = current_moment
+        db.session.commit()
+        msg = "user deleted"
+    flash(msg)
+    return redirect(url_for('user_page'))
+
+
+def is_last_admin(user):
+    admins = db.session.query(User).filter_by(role_id=ROLE_ADMIN, delete_date=None)
+    count = admins.count()
+    if count > 1:
+        return False
+    else:
+        admin = admins.first()
+        if user.id != admin.id:
+            return False
```

Comment:
Could you provide a case when we have only one admin in the system, but this is not user is going to delete him?